tasks/task_manager.hh

```diff
@@ -344,12 +354,22 @@ public:
             id = task->id();
 
             if (parent_d) {
-                co_await _tm.container().invoke_on(parent.get_owner_shard(), [task = std::move(parent), child = std::move(task)] (task_manager& tm) mutable {
-                    task->add_child(std::move(child));
-                });
+                co_await add_child(std::move(parent), std::move(task));
             }
             co_return id;
         }
+
+        // Responsibility for task_impl_ptr correctness is put on caller.
+        future<task_ptr> make_task(task::task_impl_ptr task_impl_ptr, task_info parent_d = task_info{}) {
+            auto task = make_lw_shared<task_manager::task>(std::move(task_impl_ptr));
+            foreign_task_ptr parent;
+            if (parent_d) {
+                parent = co_await get_parent(parent_d);
+                task->get_status().sequence_number = parent->get_sequence_number();
+                co_await add_child(std::move(parent), make_foreign(task));
```

Comment:
this is not very efficient since we submit_to the parent shard both in get_parent and in add_child.
if we open coded the two we could just get the parent sequence number when looking it up and update the child while at it, and eventually update the child sequnece_number, requiring only a single submit_to.
Although performance is not critical on this path, we shouldn't waste resource.

With that, the existing make_task should probably create the task_impl_ptr before looking up the parent, and then call this function to tie them together.